### PR TITLE
Fix trailing slash handling

### PR DESCRIPTION
**Pull request by guybedford**
This fixes up the resolution of specifiers with a trailing slash to ensure they skip file resolution when there exists a file by the same name.

Fixes https://github.com/vercel/node-file-trace/issues/127

### CODE REVIEW

File: src/resolve-dependency.js [hunk 1]
```diff
@@ -18,8 +21,8 @@ function resolvePath (path, parent, job) {
 }
 
 function resolveFile (path, parent, job) {
+  if (path[path.length - 1] === '/') return;
```
Comment (styfle) [inline src/resolve-dependency.js:24]: I noticed you're using `/` here but `sep` below. Why is that?

Comment (guybedford) [reply src/resolve-dependency.js:24]: In this path, it was already normalized by the line number 9 of this file to always be `/`.

File: src/resolve-dependency.js [hunk 2]
```diff
@@ -5,10 +5,13 @@ const { isAbsolute, resolve, sep } = require('path');
 // (package.json files are emitted as they are hit)
 module.exports = function resolveDependency (specifier, parent, job) {
   let resolved;
-  if (isAbsolute(specifier) || specifier === '.' || specifier === '..' || specifier.startsWith('./') || specifier.startsWith('../'))
-    resolved = resolvePath(resolve(parent, '..', specifier), parent, job);
-  else
+  if (isAbsolute(specifier) || specifier === '.' || specifier === '..' || specifier.startsWith('./') || specifier.startsWith('../')) {
+    const trailingSlash = specifier[specifier.length - 1] === '/' || specifier[specifier.lenggth - 1] === '\\';
```
Comment (styfle) [inline src/resolve-dependency.js:9]:
```suggestion
    const trailingSlash = specifier[specifier.length - 1] === '/' || specifier[specifier.length - 1] === '\\';
```

Comment (guybedford) [reply src/resolve-dependency.js]: Just saw that too, thanks for spotting.

File: src/resolve-dependency.js (hunk 1 above)
Comment (styfle) [reply src/resolve-dependency.js:24]: So should we update line 27 and 28 to use `/` instead?

Also what about line 37?

Comment (guybedford) [reply src/resolve-dependency.js:24]: 27 and 28 run against the output of realpath which is then sep-based.

For inputs into file functions, `/` and `\\` can be used interchangeably.

It's worth noting too that `require('.\\asdf')` isn't even supported in Node.js itself, so users using backslashes is already a very rare with spotty support and shouldn't be done.

**Comment by guybedford** [comment]
Test failure here seems like a flake?

File: src/resolve-dependency.js (hunk 1 above)
Comment (styfle) [reply src/resolve-dependency.js:24]: Also, why change from `path.endsWith` to  `path[path.length - 1] ===`? I think the former reads better.

```suggestion
  if (path.endsWith('/')) return;
```

File: src/resolve-dependency.js [hunk 3]
```diff
@@ -5,10 +5,13 @@ const { isAbsolute, resolve, sep } = require('path');
 // (package.json files are emitted as they are hit)
 module.exports = function resolveDependency (specifier, parent, job) {
   let resolved;
-  if (isAbsolute(specifier) || specifier === '.' || specifier === '..' || specifier.startsWith('./') || specifier.startsWith('../'))
-    resolved = resolvePath(resolve(parent, '..', specifier), parent, job);
-  else
+  if (isAbsolute(specifier) || specifier === '.' || specifier === '..' || specifier.startsWith('./') || specifier.startsWith('../')) {
+    const trailingSlash = specifier[specifier.length - 1] === '/' || specifier[specifier.length - 1] === '\\';
```
Comment (styfle) [inline src/resolve-dependency.js:9]:
```suggestion
    const trailingSlash = specifier.endsWith('/') || specifier.endsWith('\\');
```

Also, I don't see a unit test for the backslash (`\\`) use case?

Comment (guybedford) [reply src/resolve-dependency.js]: I double-checked in Node.js and it doesn't support a tailing `\`... so I've removed this test entirely.

File: src/resolve-dependency.js [hunk 4]
```diff
@@ -29,6 +32,7 @@ function resolveFile (path, parent, job) {
 }
 
 function resolveDir (path, parent, job) {
+  if (path[path.length - 1] === '/') path = path.slice(0, -1);
```
Comment (styfle) [inline src/resolve-dependency.js:35]:
```suggestion
  if (path.endsWith('/')) path = path.slice(0, -1);
```

**Comment by lukebennett** [comment]
This issue is currently blocking me from deploying a service to Vercel - any idea of how long it's likely to be until this fix trickles down to @vercel/node?

**Pull request merged**